### PR TITLE
Fix/dto and mapping for lesson

### PR DESCRIPTION
**Pull request by sober607**
Added lessons DTO

### CODE REVIEW

**Review by SerjKocherga — APPROVED** [review]

File: CharlieBackend.Core/DTO/Lesson/LessonDto.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+using System.Collections.Generic;
+using CharlieBackend.Core.DTO.Visit;
+using System.ComponentModel.DataAnnotations;
+using System.Text.Json.Serialization;
+
+namespace CharlieBackend.Core.DTO.Lesson
+{
+    public class LessonDto
+    {
+        public long Id { get; set; }
```
Comment (SnipeFrost) [inline CharlieBackend.Core/DTO/Lesson/LessonDto.cs:12]:
```suggestion
        [Required]
        public long Id { get; set; }
```
should be required

File: CharlieBackend.Core/DTO/Lesson/UpdateLessonDto.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+using CharlieBackend.Core.DTO.Visit;
+using System.Text.Json.Serialization;
+using System.ComponentModel.DataAnnotations;
+
+namespace CharlieBackend.Core.DTO.Lesson
+{
+    public class UpdateLessonDto
+    {
+        [Required]
+        [JsonIgnore]
```
Comment (SnipeFrost) [inline CharlieBackend.Core/DTO/Lesson/UpdateLessonDto.cs:13]: is it really need to use JsonIgnore?

File: CharlieBackend.Core/DTO/Visit/VisitDto.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+using System.Text;
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+using System.ComponentModel.DataAnnotations;
+
+namespace CharlieBackend.Core.DTO.Visit
+{
+    public class VisitDto
+    {
+        public long Id { get; set; }
```
Comment (SnipeFrost) [inline CharlieBackend.Core/DTO/Visit/VisitDto.cs:11]:
```suggestion
        [Required]
        public long Id { get; set; }
```

**Review by SnipeFrost — CHANGES_REQUESTED** [review]

File: CharlieBackend.Api.UnitTes/TEst.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+namespace CharlieBackend.Api.UnitTest
+{
+    public class TEst
+    {
+        [Fact]
+        public void CheckDto()
+        { 
+        var configuration = new MapperConfiguration(cfg =>
+  cfg.CreateMap<Lesson, CreateLessonDto>()
+                .ForMember(destination => destination.LessonVisits, conf => conf.MapFrom(x => x.Visits.Select(x => x.Lesson.Visits))));
```
Comment (x1mx1x) [inline CharlieBackend.Api.UnitTes/TEst.cs]: too much symbols in the line

File: CharlieBackend.Business/Services/Interfaces/ILessonService.cs [hunk 5]
```diff
@@ -1,4 +1,5 @@
-﻿using CharlieBackend.Core.Entities;
+﻿using CharlieBackend.Core.DTO.Lesson;
+using CharlieBackend.Core.Entities;
 using CharlieBackend.Core.Models.Lesson;
```
Comment (x1mx1x) [inline CharlieBackend.Business/Services/Interfaces/ILessonService.cs]: needs removing

File: CharlieBackend.Business/Services/LessonService.cs [hunk 6]
```diff
@@ -1,63 +1,67 @@
-﻿using CharlieBackend.Business.Services.Interfaces;
+﻿using System;
+using AutoMapper;
 using CharlieBackend.Core;
+using System.Threading.Tasks;
+using System.Collections.Generic;
 using CharlieBackend.Core.Entities;
+using Microsoft.AspNetCore.Diagnostics;
 using CharlieBackend.Core.Models.Lesson;
```
Comment (x1mx1x) [inline CharlieBackend.Business/Services/LessonService.cs]: needs removing

File: CharlieBackend.Api.UnitTes/TEst.cs (hunk 4 above)
Comment (sober607) [reply CharlieBackend.Api.UnitTes/TEst.cs]: TEst.cs is debug temporary file, which is absent in this pull request last commit.

File: CharlieBackend.Core/DTO/Lesson/UpdateLessonDto.cs (hunk 2 above)
Comment (x1mx1x) [inline CharlieBackend.Core/DTO/Lesson/UpdateLessonDto.cs:13]: remove JsonIgnore

File: CharlieBackend.Core/Entities/Lesson.cs [hunk 7]
```diff
@@ -24,6 +21,6 @@ public Lesson()
 
         public virtual Theme Theme { get; set; }
         
-        public virtual ICollection<Visit> Visits { get; set; }
+        public virtual IList<Visit> Visits { get; set; }
```
Comment (x1mx1x) [inline CharlieBackend.Core/Entities/Lesson.cs:23]: Are you sure that it is a good solution?

Comment (sober607) [reply CharlieBackend.Core/Entities/Lesson.cs:23]: Ilist gives more possibilities foreach loop or a for loop to iterate

**Review by x1mx1x — APPROVED** [review]
ok

**Review by SnipeFrost — APPROVED** [review]
good

**Review by BohdanBBG — APPROVED** [review]

**Review by Vlad1023 — APPROVED** [review]

**Pull request merged**